networking/p2p/src/integration-test/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/GossipMessageHandlerIntegrationTest.java

```diff
@@ -88,6 +105,116 @@ public void shouldGossipBlocksAcrossToIndirectlyConnectedPeers() throws Exceptio
+        });
+    // TODO: debug this - we shouldn't have to wait here
+    Thread.sleep(2000);
+
+    // Listen for new block event to arrive on network 2
+    final AttestationCollector network2Attestations = new AttestationCollector(eventBus2);
+
+    // Propagate attestation from network 1
+    AttestationGenerator attestationGenerator = new AttestationGenerator(blsKeyPairList);
+    Attestation validAttestation = attestationGenerator.validAttestation(storageClient1);
+    eventBus1.post(validAttestation);
+
+    Thread.sleep(2000);
+
+    assertThat(network2Attestations.getAttestations()).isEmpty();
+  }
+
+  @Test
+  public void shouldReceiveAttestationAcrossPeersInSameSubnet() throws Exception {
+    final int numValidators = 12;
+
+    ArtemisConfiguration config = mock(ArtemisConfiguration.class);
+    doReturn(0).when(config).getNaughtinessPercentage();
+    doReturn(numValidators).when(config).getNumValidators();
+    doReturn(null).when(config).getValidatorsKeyFile();
+    doReturn(0).when(config).getInteropOwnedValidatorStartIndex();
+    doReturn(numValidators).when(config).getInteropOwnedValidatorCount();
+
+    AttestationAggregator aggregator = mock(AttestationAggregator.class);
+    doNothing().when(aggregator).updateAggregatorInformations(any());
+
+    // Setup network 1
+    final EventBus eventBus1 = new EventBus();
+    final ChainStorageClient storageClient1 = new ChainStorageClient(eventBus1);
+    final JvmLibP2PNetwork network1 = networkFactory.startNetwork(eventBus1, storageClient1);
+    List<BLSKeyPair> blsKeyPairList =
+        new MockStartValidatorKeyPairFactory().generateKeyPairs(0, numValidators);
+    final BeaconChainUtil chainUtil = BeaconChainUtil.create(storageClient1, blsKeyPairList);
+    chainUtil.initializeStorage();
+
+    // Setup network 2
+    final EventBus eventBus2 = new EventBus();
+    final ChainStorageClient storageClient2 = new ChainStorageClient(eventBus2);
+    final JvmLibP2PNetwork network2 = networkFactory.startNetwork(eventBus2, storageClient2);
+    chainUtil.initializeStorage(storageClient2);
+
+    // Setup VC 1
+    new ValidatorCoordinator(
+        eventBus1, storageClient1, aggregator, mock(BlockAttestationsPool.class), config);
+    eventBus1.post(new StoreInitializedEvent());
+
+    // Setup VC 2
+    Constants.VALIDATOR_CLIENT_PORT_BASE = Constants.VALIDATOR_CLIENT_PORT_BASE + numValidators;
+    new ValidatorCoordinator(
+        eventBus2, storageClient2, aggregator, mock(BlockAttestationsPool.class), config);
+    eventBus2.post(new StoreInitializedEvent());
+
+    // Connect networks 1 -> 2
+    network1.connect(network2.getPeerAddress());
+    // Wait for connections to get set up
+    Waiter.waitFor(
+        () -> {
+          assertThat(network1.getPeerManager().getAvailablePeerCount()).isEqualTo(1);
+          assertThat(network2.getPeerManager().getAvailablePeerCount()).isEqualTo(1);
+        });
+
+    Thread.sleep(500);
```

Comment:
Why do we need the sleeps?

Reply:
I explained in comments why I put sleeps.